### PR TITLE
feat(consumer-group list): add search flag

### LINKED ISSUE #714: Add search flag for `kafka consumergroup list`
### Feature or problem description

User should be able to filter consumer groups by id.

### Proposed solution

Add a --search flag for `rhoas kafka consumergroup list` to filter topics by name.

```
rhoas kafka consumergroup list --search "<search input>"
```

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Add `--search` flag for consumer group list.

Closes #714 

### Verification Steps

1. Create consumer groups for a Kafka instance.
2. Run consumer group list command with search flag.
```
go run ./cmd/rhoas kafka consumer-group list --search console-consume -o yaml
```
3. It should return the consumer groups.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [X] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/kafka/consumergroup/validators.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/redhat-developer/app-services-cli/pkg/common/commonerr"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+)
+
+const (
```
Comment (rkpattnaik780) [inline pkg/kafka/consumergroup/validators.go:11]: Should this be correct regexp? ( alphanumeric and hyphen).

File: pkg/kafka/consumergroup/validators.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	legalSearchChars = "^[a-zA-Z0-9-]+$"
```
Comment (wtrocki) [inline pkg/kafka/consumergroup/validators.go:12]: What validation for search you have in backend @sknot-rh @MikeEdgar ?

File: pkg/kafka/consumergroup/validators.go (hunk 1 above)
Comment (wtrocki) [reply pkg/kafka/consumergroup/validators.go:11]: Asked in like that contains regexp

File: pkg/kafka/consumergroup/validators.go (hunk 2 above)
Comment (MikeEdgar) [reply pkg/kafka/consumergroup/validators.go:12]: Which search filter does this apply to? The `topic` query param?

Comment (rkpattnaik780) [reply pkg/kafka/consumergroup/validators.go:12]: It is for `group-id-filter` param.

Comment (MikeEdgar) [reply pkg/kafka/consumergroup/validators.go:12]: @rkpattnaik780 , @wtrocki - there is currently no validation for that param.

Comment (rkpattnaik780) [reply pkg/kafka/consumergroup/validators.go:12]: @wtrocki @MikeEdgar @craicoverflow should we restrict special characters for this argument? Passing `*` gets into filesystem.
Screenshot from 2021-07-09 12-53-06

Comment (rkpattnaik780) [reply pkg/kafka/consumergroup/validators.go:12]: @craicoverflow turns out validators wont prevent us from this glitch, user needs to use double quotes. Shall I get rid of validation here altogether?
Screenshot from 2021-07-09 16-31-47

Comment (craicoverflow) [reply pkg/kafka/consumergroup/validators.go:12]: Ah I see. yes do, thanks.

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**